### DISCUSSION TITLE
Permanent Delete on OneDrive?

### DISCUSSION THREAD

**Discussion by NovaPixell** [abraunegg/onedrive#2802] [Q&A] [answered]
Im using onedrive v2.4.25-1+np1 on my NAS (OpenMediaVault).

### OneDrive Config
```
root@nas:~# cat /var/cache/onedrive/config 
# This file is auto-generated by openmediavault (https://www.openmediavault.org)
# WARNING: Do not edit this file, your changes will get lost.
enable_logging = "false"
verbose = "0"
sync_dir = "/srv/dev-disk-by-uuid-a3e7c153-0ea5-4785-ba17-615d15177cac/DATI/"
sync_dir_permissions = "750"
sync_file_permissions = "640"
disable_notifications = "true"
local_first = "false"
no_remote_delete = "false"
download_only = "false"
upload_only = "true"
skip_dotfiles = "false"
monitor_interval = "86400"
monitor_fullscan_frequency = "12"
rate_limit = "0"
```
### Question:
Is it possible to enable permanent delete on OneDrive? I am using OneDrive Business with Office 365, whenever a file is deleted locally and also on OneDrive, it is not permanently deleted but is moved to the trash bin. 

Is there an option (like on rclone) to enable permanent delete on OneDrive?

**Comment by abraunegg** [comment, marked as answer]
@NovaPixell 
> Is there an option (like on rclone) to enable permanent delete on OneDrive?

Unfortunately no such option exists currently. Please raise a feature request.

Additionally, due to the current development cycle, this will be looked at for some version in v2.5.x and cannot be considered for inclusion in v2.5.0.

**Reference:**
* https://learn.microsoft.com/en-us/graph/api/driveitem-permanentdelete?view=graph-rest-1.0
